staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/testing/promise/counting_test.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package promise
+
+import (
+	"context"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/util/flowcontrol/counter"
+	"k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/promise"
+	testeventclock "k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/testing/eventclock"
+	"k8s.io/klog/v2"
+)
+
+func TestMain(m *testing.M) {
+	klog.InitFlags(nil)
+	os.Exit(m.Run())
+}
+
+func TestCountingWriteOnceSet(t *testing.T) {
+	oldTime := time.Now()
+	cval := &oldTime
+	doneCh := make(chan struct{})
+	now := time.Now()
+	clock, counter := testeventclock.NewFake(now, 0, nil)
+	var lock sync.Mutex
+	wr := NewCountingWriteOnce(counter, &lock, nil, doneCh, cval)
+	gots := make(chan interface{}, 1)
+	counter.Add(1)
+	go func() {
+		defer counter.Add(-1)
+		gots <- wr.Get()
+	}()
+	clock.Run(nil)
+	select {
```

Comment:
Same two comments as in the other test:
- let's also extract expectNoGotValue to a function
- let's reduce the timeout to 1s